cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +70,33 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def find_reachable_deps(yarn_lock, dep):
+    """
+    Get a set of all dependencies reachable by a top-level dependency using BFS.
+
+    :param yarn_lock: yarn.lock file as a dictionary
+    :param dep: top-level dependency in package.json
+    :return: set of all dependencies reachable by dep
+    :rtype: set
+    """
+    yarn_lock_parsed = _expand_replacements(yarn_lock)
+    bfs_queue = deque([dep])
+    reachable_deps = set([dep])
+    while bfs_queue:
+        current_dep = bfs_queue.popleft()
```

Comment:
```suggestion
        current_dep = bfs_queue.popleft()
        reachable_deps.add(current_dep)
```

Reply:
`reachable_deps.add(current_dep)` should still be in line 95 because of line 92.

Reply:
Hm, are you sure? In my opinion, they don't have any correlation. `current_dep` is a node in our "graph". If we visit this node, we need to add it to our result. 

in line 92 you check children of `current_dep`. We don't need to add them to our queue in case we already visited them.

Reply:
But it doesn't matter actually! The time complexity is already better :)